jupyterhub/apihandlers/users.py

```diff
@@ -371,12 +382,17 @@ def delete(self, name, token_id):
 class UserServerAPIHandler(APIHandler):
     """Start and stop single-user servers"""
 
-    @admin_or_self
-    async def post(self, name, server_name=''):
+    @needs_scope('users:servers')
+    async def post(self, name, server_name='', subset=None):
```

Comment:
apply resource filtering in decorator, e.g. explicitly:

```python
@needs_scope("users:servers", user_pos=0, server_pos=1)
```


or automatic (at least default) by inspection, i.e. `name` == username, `server_name` == server name, etc.